### PR TITLE
Require `libgomp` on Linux

### PR DESCRIPTION
**Pull request by jakirkham**
Looks like OpenMP is used for the builds on macOS, but the dependency is not included on Linux. This adds `libgomp` on Linux to provide OpenMP support on that platform.

<hr>

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jjerphan** [comment]
Hi @jakirkham, thank you for contributing.

Have you observed any problems at runtime with the current build (e.g. no thread parallelism?).

**Comment by jakirkham** [comment]
Thanks Julien! 🙏

A colleague was seeing a rather strange error message in relation to scinkit-learn usage ( https://github.com/dask-contrib/dask-sql/pull/1144#issuecomment-1559791735 ), which when digging more deeply can be caused by GNU OpenMP either being missing or not found correctly ( for example: https://github.com/opencv/opencv/issues/14884#issuecomment-1114003997 )

As LLVM OpenMP is included here for macOS and [the usual recommendation]( https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp ) suggests requiring GNU OpenMP on Linux too, thought we could give this change a try

It's possible the issue encountered has other causes, but this would at least rule out one (while improving the package here as well)

Would be curious to hear your thoughts on all of this? 🙂

**Review by jjerphan — APPROVED** [review]
Thank you for providing context.

scikit-learn's builds for Linux distributed on conda-forge has been depending on libgomp implicitly since the support of OpenMP but I think that it is appropriate specifying it explicitly for portability (since it allows switching to `llvm-openmp` if needed).

I am waiting for another maintainer's approval before merging.

**Review by thomasjpfan — APPROVED** [review]
This change makes sense to me. LGTM!

**Comment by jakirkham** [comment]
Thanks Julien & Thomas! 🙏

**Comment by jjerphan** [comment]
Thank you for resolving problems across packages in the ecosystem, John.

I hope to see you at the Scientific Python Developer Summit. :deciduous_tree:

**Comment by jakirkham** [comment]
Ofc!

Hoping to come over there today 😄

**Pull request merged**
